From 2a965feea818bc194b74383e504608566167fbe0 Mon Sep 17 00:00:00 2001
From: linmin <linmin@eswincomputing.com>
Date: Mon, 17 Jun 2024 14:26:26 +0800
Subject: [PATCH 092/109] WIN2030-15279:fix(power consumption):Bug fix for
 get_board_power()

Changelogs:
1.Deleted the taskENTER_CRITICAL()/taskEXIT_CRITICAL() in get_som_power().
  Because this will disable interrupt, then the tick will not be
  increased, so that the i2c api will never timeout and will wait forever.
2.Updated the BMC version to 2.1

Change-Id: I9a01fa025d82dcb9a8bd0784067e414440d2bbe5
---
 Core/Inc/hf_common.h        | 2 +-
 Core/Src/hf_power_process.c | 2 --
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/Core/Inc/hf_common.h b/Core/Inc/hf_common.h
index 0f43eed..9c9bb5a 100644
--- a/Core/Inc/hf_common.h
+++ b/Core/Inc/hf_common.h
@@ -217,7 +217,7 @@ extern UART_HandleTypeDef huart3;
 #define MCU_MAC_IDX	2
 /* define ------------------------------------------------------------*/
 #define BMC_SOFTWARE_VERSION_MAJOR                   2
-#define BMC_SOFTWARE_VERSION_MINOR                   0
+#define BMC_SOFTWARE_VERSION_MINOR                   1
 
 #define MAGIC_NUMBER	0x45505EF1
 
diff --git a/Core/Src/hf_power_process.c b/Core/Src/hf_power_process.c
index a6b4dac..29b1c46 100644
--- a/Core/Src/hf_power_process.c
+++ b/Core/Src/hf_power_process.c
@@ -331,7 +331,6 @@ int get_board_power(uint32_t *volt, uint32_t *curr, uint32_t *power)
 
 	es_get_rtc_time(&time);
 
-	taskENTER_CRITICAL();
 	if (1 == is_first)
 	{
 		/*Write failure, not find wrong reason*/
@@ -376,7 +375,6 @@ int get_board_power(uint32_t *volt, uint32_t *curr, uint32_t *power)
 	*curr = DIV_ROUND_CLOSEST(*curr, 1000);
 
 out:
-	taskEXIT_CRITICAL();
 	return ret;
 }
 
-- 
2.25.1

